cpp/oneapi/dal/algo/knn/common.hpp

```diff
@@ -71,6 +75,43 @@ using v1::by_default;
 
 } // namespace method
 
+namespace result_options {
+
+/// Represents result option flag
+/// Behaves like a regular :expr`enum`.
+struct result_option_id_t : public result_option_id {
+    result_option_id_t() : result_option_id{} {}
+    result_option_id_t(const result_option_id& base) : result_option_id{ base } {}
+};
+
+} // namespace result_options
+
+namespace detail {
+
+/// Alias for :expr`result_options::result_option_id_t`
+using result_option_id_t = result_options::result_option_id_t;
+ONEDAL_EXPORT result_option_id_t get_indices_id();
+ONEDAL_EXPORT result_option_id_t get_distances_id();
+ONEDAL_EXPORT result_option_id_t get_responses_id();
+
+} // namespace detail
+
+/// Result options are used to define
+/// what should algorithm return
+namespace result_options {
+
+const inline result_option_id_t indices = detail::get_indices_id();
+const inline result_option_id_t distances = detail::get_distances_id();
+const inline result_option_id_t responses = detail::get_responses_id();
+
+template <typename Task>
+const inline result_option_id_t default_result_options = responses;
+
+template <>
+const inline result_option_id_t default_result_options<task::search> = indices | distances;
```

Comment:
I don't remember we approved this kind of public API. To me, this should not be exposed in the header files.

Reply:
Fixed